CONTRIBUTING.md

```diff
@@ -4,4 +4,97 @@ Thank you for your interest in the Langium project! The following is a set of gu
 
 ## Prerequisites
 
-Developing Langium only requires `npm@^7.7.0`. For this you will need `Node.js 10` or greater.
\ No newline at end of file
+For developing Langium you require at least the Maintenance LTS version of Node.js (currently 12) and at least npm version 7.7.0 (`npm@^7.7.0`) to be able to use npm workspaces. With Node.js versions below 16 you have to install the correct version of npm with `npm install -g npm@7`.
+
+
+## Local Development Guide
+
+We outline three use-cases:
+1. Developing Langium itself
+2. Developing you own language and use Langium as tool
+3. Developing Langium itself and developing your own language itself.
+
+The first two use-cases are the common ones. The third one is less common, but there are some recommendations you should know.
+
+### Developing Langium
+
+Development of Langium does not necessarily require VSCode (https://code.visualstudio.com/download), but without it you will not be able to use language server and its extension for VSCode.
+Langium is an npm workspace project with currently four main packages (`langium`, `langium-cli`, `langium-vscode` and `generator-langium`).
```

Comment:
Writing details such as the number and the names of contained packages here means we have to maintain this whenever there are changes. I'd rather shorten this to "multiple packages", and maybe link that to the `packages` folder in GitHub.

Reply:
Good idea. I will change it accordingly.